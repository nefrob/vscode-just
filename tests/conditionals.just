# Conditional expressions

foo := if "2" == "2" { "Good!" } else { "1984" }
foo := if "hello" != "goodbye" { "xyz" } else { "abc" }
foo := if "hello" =~ 'hel+o' { "match" } else { "mismatch" }
foo := if env_var("RELEASE") == "true" { `get-something-from-release-database` } else { "dummy-value" }
foo := if "hello" == "goodbye" {
  "xyz"
} else if "a" == "a" {
  "abc"
} else {
  "123"
}