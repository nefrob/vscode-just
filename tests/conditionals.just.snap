># Conditional expressions
#^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>foo := if "2" == "2" { "Good!" } else { "1984" }
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^ source.just string.quoted.double.just
#           ^ source.just string.quoted.double.just
#            ^ source.just string.quoted.double.just
#             ^ source.just
#              ^^ source.just keyword.operator.equality.just
#                ^ source.just
#                 ^ source.just string.quoted.double.just
#                  ^ source.just string.quoted.double.just
#                   ^ source.just string.quoted.double.just
#                    ^^^ source.just
#                       ^ source.just string.quoted.double.just
#                        ^^^^^ source.just string.quoted.double.just
#                             ^ source.just string.quoted.double.just
#                              ^^^ source.just
#                                 ^^^^ source.just keyword.control.conditional.just
#                                     ^^^ source.just
#                                        ^ source.just string.quoted.double.just
#                                         ^^^^ source.just string.quoted.double.just
#                                             ^ source.just string.quoted.double.just
#                                              ^^^ source.just
>foo := if "hello" != "goodbye" { "xyz" } else { "abc" }
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^ source.just string.quoted.double.just
#           ^^^^^ source.just string.quoted.double.just
#                ^ source.just string.quoted.double.just
#                 ^ source.just
#                  ^^ source.just keyword.operator.equality.just
#                    ^ source.just
#                     ^ source.just string.quoted.double.just
#                      ^^^^^^^ source.just string.quoted.double.just
#                             ^ source.just string.quoted.double.just
#                              ^^^ source.just
#                                 ^ source.just string.quoted.double.just
#                                  ^^^ source.just string.quoted.double.just
#                                     ^ source.just string.quoted.double.just
#                                      ^^^ source.just
#                                         ^^^^ source.just keyword.control.conditional.just
#                                             ^^^ source.just
#                                                ^ source.just string.quoted.double.just
#                                                 ^^^ source.just string.quoted.double.just
#                                                    ^ source.just string.quoted.double.just
#                                                     ^^^ source.just
>foo := if "hello" =~ 'hel+o' { "match" } else { "mismatch" }
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^ source.just string.quoted.double.just
#           ^^^^^ source.just string.quoted.double.just
#                ^ source.just string.quoted.double.just
#                 ^ source.just
#                  ^^ source.just keyword.operator.equality.just
#                    ^ source.just
#                     ^ source.just string.quoted.single.just
#                      ^^^^^ source.just string.quoted.single.just
#                           ^ source.just string.quoted.single.just
#                            ^^^ source.just
#                               ^ source.just string.quoted.double.just
#                                ^^^^^ source.just string.quoted.double.just
#                                     ^ source.just string.quoted.double.just
#                                      ^^^ source.just
#                                         ^^^^ source.just keyword.control.conditional.just
#                                             ^^^ source.just
#                                                ^ source.just string.quoted.double.just
#                                                 ^^^^^^^^ source.just string.quoted.double.just
#                                                         ^ source.just string.quoted.double.just
#                                                          ^^^ source.just
>foo := if env_var("RELEASE") == "true" { `get-something-from-release-database` } else { "dummy-value" }
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^^^^^^^ source.just support.function.builtin.just
#                 ^ source.just
#                  ^ source.just string.quoted.double.just
#                   ^^^^^^^ source.just string.quoted.double.just
#                          ^ source.just string.quoted.double.just
#                           ^ source.just
#                            ^ source.just
#                             ^^ source.just keyword.operator.equality.just
#                               ^ source.just
#                                ^ source.just string.quoted.double.just
#                                 ^^^^ source.just string.quoted.double.just
#                                     ^ source.just string.quoted.double.just
#                                      ^^^ source.just
#                                         ^ source.just string.interpolated.just
#                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just source.shell
#                                                                             ^ source.just string.interpolated.just
#                                                                              ^^^ source.just
#                                                                                 ^^^^ source.just keyword.control.conditional.just
#                                                                                     ^^^ source.just
#                                                                                        ^ source.just string.quoted.double.just
#                                                                                         ^^^^^^^^^^^ source.just string.quoted.double.just
#                                                                                                    ^ source.just string.quoted.double.just
#                                                                                                     ^^^ source.just
>foo := if "hello" == "goodbye" {
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^ source.just string.quoted.double.just
#           ^^^^^ source.just string.quoted.double.just
#                ^ source.just string.quoted.double.just
#                 ^ source.just
#                  ^^ source.just keyword.operator.equality.just
#                    ^ source.just
#                     ^ source.just string.quoted.double.just
#                      ^^^^^^^ source.just string.quoted.double.just
#                             ^ source.just string.quoted.double.just
#                              ^^^ source.just
>  "xyz"
#^^ source.just
#  ^ source.just string.quoted.double.just
#   ^^^ source.just string.quoted.double.just
#      ^ source.just string.quoted.double.just
>} else if "a" == "a" {
#^^ source.just
#  ^^^^ source.just keyword.control.conditional.just
#      ^ source.just
#       ^^ source.just keyword.control.conditional.just
#         ^ source.just
#          ^ source.just string.quoted.double.just
#           ^ source.just string.quoted.double.just
#            ^ source.just string.quoted.double.just
#             ^ source.just
#              ^^ source.just keyword.operator.equality.just
#                ^ source.just
#                 ^ source.just string.quoted.double.just
#                  ^ source.just string.quoted.double.just
#                   ^ source.just string.quoted.double.just
#                    ^^^ source.just
>  "abc"
#^^ source.just
#  ^ source.just string.quoted.double.just
#   ^^^ source.just string.quoted.double.just
#      ^ source.just string.quoted.double.just
>} else {
#^^ source.just
#  ^^^^ source.just keyword.control.conditional.just
#      ^^^ source.just
>  "123"
#^^ source.just
#  ^ source.just string.quoted.double.just
#   ^^^ source.just string.quoted.double.just
#      ^ source.just string.quoted.double.just
>}
#^^ source.just